resources/repo.rb

```diff
@@ -13,6 +14,7 @@
     repo_type config['recipeName']
     attributes config['attributes']
     online config['online']
+    routing_rule_name config['routingRuleName']
```

Comment:
If I'm not mistaken your new version of `get_repo` is returning `null` when no routing rule has been setup; so this is going to fail for repository already created with no routing rule as you only accept `String`.

Reply:
No it returns a string.

Reply:
(An empty string)

Reply:
For posterity, value was actually `null`, as groovy does not initialize a value by default ` String s;`, `s` is null. But as stated in the other comment this is fine, default value will be used.

Reply:
After testing it seems in some case the script returns `null` but it's OK as the commented line is valid even if the value is `nil`.